### PR TITLE
JAMES-3395 make examples not inherit from parent, move package of examples out from org.apache.james

### PR DESCRIPTION
**Pull request by aasaru**

### CODE REVIEW

**Comment by Arsnael** [comment]
Thanks for the initiative :) I don't think I have issues with that, on the contrary

Just a question, is `org.apache.examples.james` as a groupId a good one? (Opened question, I don't pretend having the answer)

**Comment by aasaru** [comment]
one alternative to that would be org.example.james

File: examples/pom.xml [hunk 1]
```diff
@@ -36,4 +33,14 @@
         <module>custom-mailets</module>
     </modules>
 
-</project>
\ No newline at end of file
+    <properties>
+        <james.groupId>org.apache.james</james.groupId>
+        <james.baseVersion>3.6.0-SNAPSHOT</james.baseVersion>
```
Comment (chibenwa) [inline examples/pom.xml:38]: I understand the concerns however it likely makes the release process hard.

How do we ensure the version use by the example matches the released version? How does it integrate with the maven-release plugin?

I I was to tag a 3.6.0 tomorrow, I would likely forget that. Even if I don't forget, how do I ensure the released exemple JAR matches the 3.6.0 version? Would I have to do a pre-release commit switching it to the not yet released 3.6.0 - in which case the whole process might fail at publishing snapshots?

Maybe a wiser value here would be :

```
<james.baseVersion>${project.version}</james.baseVersion>
```

That way as the version of the example and James are implicitly aliagned, the maven-release prompt will enable me to do the switch painlessly...

**Comment by Arsnael** [comment]
I don't think you can omit `apache` in there, as those modules belong to the apache foundation right?

File: examples/pom.xml [hunk 2]
```diff
@@ -20,14 +20,11 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <parent>
-        <groupId>org.apache.james</groupId>
-        <artifactId>james-project</artifactId>
-        <version>3.6.0-SNAPSHOT</version>
-    </parent>
+    <groupId>org.apache.examples.james</groupId>
```
Comment (chibenwa) [inline examples/pom.xml:23]: I'm unsure that the James project is allowed to publish JARs to maven central  with a `org.apache.examples.` prefix. 

Can you check the Apache foundation policies regarding this?

**Review by chibenwa — COMMENTED** [review]
Globally agree, though release details needs to be sorted out.

File: examples/pom.xml (hunk 2 above)
Comment (chibenwa) [reply examples/pom.xml:23]: https://infra.apache.org/publishing-maven-artifacts.html don't seem helpful regarding this concern...

Comment (chibenwa) [reply examples/pom.xml:23]: Neither do https://infra.apache.org/release-distribution.html ...

Comment (chibenwa) [reply examples/pom.xml:23]: Maybe we need to ask the INFRA team?

Comment (chibenwa) [reply examples/pom.xml:23]: Also using org.apache.examples prefix don't seem to be a standard practice amongst other project.

See https://mvnrepository.com/search?q=org.apache.examples

Maybe we should keep the `org.apache.james.examples` FQDN while switching it to a top level POM not depending on James prent POM. ?

File: examples/pom.xml [hunk 3]
```diff
@@ -20,14 +20,11 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <parent>
-        <groupId>org.apache.james</groupId>
-        <artifactId>james-project</artifactId>
-        <version>3.6.0-SNAPSHOT</version>
-    </parent>
```
Comment (chibenwa) [inline examples/pom.xml:27]: We at least need to depend on the root Apache POM as a parent .

See https://infra.apache.org/publishing-maven-artifacts.html

File: examples/pom.xml (hunk 1 above)
Comment (aasaru) [reply examples/pom.xml]: Thanks for good feedback. I didn't release the examples are get published so of course it makes sense to use this property as a default.

File: examples/pom.xml (hunk 2 above)
Comment (aasaru) [reply examples/pom.xml]: After thinking about this I realised that if we have all example code under org.apache.james.examples then we cannot use package level access anywhere (as core james won't have the 'examples' part in the package name). Sorry for creating confusion - reverted back this part - let's stay to org.apache.james.examples

File: examples/pom.xml (hunk 3 above)
Comment (aasaru) [reply examples/pom.xml:27]: Good point. Made root apache pom as parent.

File: README.adoc [hunk 4]
```diff
@@ -401,7 +401,8 @@ Beware : you will have concurrency issues if multiple containers are running on
 
 == Develop on James
 
-James requires at least JDK 11 and maven 3.6.0 to build.
+James requires at least JDK 11 and Maven 3.6.0 to build.
+Some parts of James are written in Scala so one might need to enable Scala plugin in IDE.
```
Comment (chibenwa) [inline README.adoc:405]: <3

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged. Can you please help close the PR @aasaru ? Thank you very much for your contribution :)

**Pull request closed without merging**